chips/esp32/src/uart.rs

```diff
@@ -330,6 +361,27 @@ impl<'a> Uart<'a> {
                 self.tx_progress();
             }
         }
+
+        if intrs.is_set(INT::RXFIFO_FULL_INT) || intrs.is_set(INT::RXFIFO_TOUT_INT) {
+            self.disable_rx_interrupt();
```

Comment:
I think this line should  be moved inside the if statement on the next line. There is no reason to disable interrupts if we have not received everything, and disabling here then re-enabling in `rx_progress()` wastes cycles and risks missing interrupts that arrive in that brief window of time.

Reply:
I just replicated what @alistair23 did for the transmission part, but you have a point. I moved both disabling of interrupts just before running the client's callback function.